### ISSUE TITLE
Circular dependency detection does not work correctly

### ISSUE THREAD

**Issue by karelkral** [radzserg/rsdi-legacy#6]
Hello, just upgraded to a 1.0.9 version of your DI Container and my application fails with this error:

> Error: Circular Dependency is detected. Dependency: "AxiosClientProvider", path: TfsRepositoryService -> AxiosClientProvider -> SdkService -> SharedDataService.

But as I tried to resolve this, I think that this error is a false positive. I didn't go too deep into your sources but I think that this is not a circular dependency but a bug in detection.

So for now I have to revert to a previous version of the library.

My dependencies are bellow:
```
    container.addDefinition(ServiceId.TfsRepositoryService, object(TfsRepositoryDataService).construct(
		get(ServiceId.AxiosClientProvider), get(ServiceId.SharedDataService)
	));
	container.addDefinition(ServiceId.AxiosClientProvider, object(AxiosClientProvider).construct(
		get(ServiceId.ConfigService), get(ServiceId.SdkService)
	));
	container.addDefinition(ServiceId.SharedDataService, object(SharedDataService).construct(
		get(ServiceId.AxiosClientProvider)
	));
	container.addDefinition(ServiceId.ConfigService, object(ConfigService));
	container.addDefinition(ServiceId.SdkService, object(SdkService));

  export class ServiceId {
      public static readonly AxiosClientProvider: string = "AxiosClientProvider";
      public static readonly ConfigService: string = "ConfigService";
      public static readonly SdkService: string = "SdkService";
      public static readonly SharedDataService: string = "SharedDataService";
      public static readonly TfsRepositoryService: string = "TfsRepositoryService";
}

```

**Comment by radzserg** [comment]
Hey @karelkral I made a fix, can you try v1.0.10

**Comment by karelkral** [comment]
1.0.10 is working, thanks for a fast fix!
